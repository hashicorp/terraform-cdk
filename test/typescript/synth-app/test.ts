// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import { TestDriver } from "../../test-helper";

describe("full integration test synth", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    process.env.TF_VAR_myvar =
      "If you see this value we pass runtime information into the synth";
    driver = new TestDriver(__dirname);
    await driver.setupTypescriptProject();
    await driver.synth();
  });

  test("debug command", async () => {
    const { stdout } = await driver.exec(`cdktf debug --json`);
    const { cdktf, constructs } = JSON.parse(stdout);
    expect(cdktf.length).not.toBe(0);
    expect(constructs.length).not.toBe(0);
  });

  test("synth generates JSON", () => {
    expect(driver.synthesizedStack("hello-terra").toString()).toMatchSnapshot();
  });

  test("should not pass TF_VAR environment variables", () => {
    expect(
      driver.synthesizedStack("hello-terra").output("tf-env-var-output")
    ).toBe("no-value-found");
  });
});
