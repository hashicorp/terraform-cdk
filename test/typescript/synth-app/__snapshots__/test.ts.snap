// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full integration test synth synth generates JSON 1`] = `
"{
  "//": {
    "metadata": {
      "backend": "local",
      "overrides": {
        "aws_sns_topic": [
          "display_name",
          "provider",
          "lifecycle"
        ],
        "stack": [
          "terraform"
        ]
      },
      "stackName": "hello-terra",
      "version": "stubbed"
    },
    "outputs": {
      "hello-terra": {
        "instance-http-endpoint": "instance-http-endpoint",
        "instance-password": "instance-password",
        "sns-topic-arn": "sns-topic-arn",
        "tf-env-var-output": "tf-env-var-output"
      }
    }
  },
  "output": {
    "instance-http-endpoint": {
      "value": "\${aws_instance.Instance.metadata_options[0].http_endpoint}"
    },
    "instance-password": {
      "value": "\${base64decode(aws_instance.Instance.password_data)}"
    },
    "sns-topic-arn": {
      "value": "\${aws_sns_topic.Topic.arn}"
    },
    "tf-env-var-output": {
      "value": "no-value-found"
    }
  },
  "provider": {
    "aws": [
      {
        "ignore_tags": {
          "keys": [
            "foo"
          ]
        },
        "region": "eu-central-1"
      }
    ]
  },
  "resource": {
    "aws_instance": {
      "Instance": {
        "//": {
          "metadata": {
            "path": "hello-terra/Instance",
            "uniqueId": "Instance"
          }
        },
        "ami": "ami-12345678",
        "credit_specification": {
          "cpu_credits": "standard"
        },
        "ebs_block_device": [
          {
            "device_name": "/dev/sda1",
            "volume_size": 100
          }
        ],
        "instance_type": "t2.micro",
        "metadata_options": {
          "http_endpoint": "true"
        }
      },
      "Instance2": {
        "//": {
          "metadata": {
            "path": "hello-terra/Instance2",
            "uniqueId": "Instance2"
          }
        },
        "ami": "ami-12345678",
        "credit_specification": {
          "cpu_credits": "\${aws_instance.Instance.credit_specification[0].cpu_credits}"
        },
        "instance_type": "t2.micro",
        "metadata_options": {
          "http_endpoint": "\${aws_instance.Instance.metadata_options[0].http_endpoint}"
        }
      }
    },
    "aws_sns_topic": {
      "Topic": {
        "//": {
          "metadata": {
            "path": "hello-terra/Topic",
            "uniqueId": "Topic"
          }
        },
        "display_name": "topic",
        "lifecycle": {
          "create_before_destroy": true
        },
        "provider": "aws"
      }
    },
    "aws_wafv2_web_acl": {
      "wafv2": {
        "//": {
          "metadata": {
            "path": "hello-terra/wafv2",
            "uniqueId": "wafv2"
          }
        },
        "default_action": {
          "allow": {}
        },
        "name": "managed-rule-example",
        "rule": [
          {
            "name": "managed-rule-example",
            "override_action": {
              "count": {}
            },
            "priority": 1,
            "statement": {
              "managed_rule_group_statement": {
                "excluded_rule": [
                  {
                    "name": "SizeRestrictions_QUERYSTRING"
                  },
                  {
                    "name": "SQLInjection_QUERYSTRING"
                  }
                ],
                "name": "managed-rule-example",
                "vendor_name": "AWS"
              }
            },
            "visibility_config": {
              "cloudwatch_metrics_enabled": true,
              "metric_name": "managed-rule-example",
              "sampled_requests_enabled": true
            }
          }
        ],
        "scope": "REGIONAL",
        "visibility_config": {
          "cloudwatch_metrics_enabled": true,
          "metric_name": "managed-rule-example",
          "sampled_requests_enabled": true
        }
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "terraform.tfstate"
      },
      "remote": {
        "organization": "test",
        "workspaces": {
          "name": "test"
        }
      }
    },
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "2.70.4"
      }
    }
  }
}"
`;
